ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -321,17 +321,18 @@ internal constructor(
     if (tasks.isEmpty()) {
       error("Can't generate sequence for empty task list")
     }
-
-    val task = tasks.filter { it.id == (startId ?: tasks[0].id) }
-
-    // TODO(#2539): Cleanup once https://github.com/google/ground-android/issues/2539 is resolved.
-    if (task.isEmpty()) {
-      error(
-        "Unable to find a task with id startId=$startId, firstTaskId=${tasks[0].id}, allTasks=${tasks.map { it.id }}"
-      )
-    }
-
-    val startIndex = tasks.indexOf(task.first())
+    val startIndex =
+      tasks
+        .indexOfFirst { it.id == (startId ?: tasks[0].id) }
```

Comment:
Can we move `tasks[0].id` outside of the filtering logic?

Reply:
Done.